lib/account.rb

```diff
@@ -0,0 +1,168 @@
+      # Looks in @accounts array for a matching ID value
+      found = @accounts.find do |account|
+        account.id == id_search.to_s
+      end
+      return found
+    end
+
+    # Links owners with accounts
+    def self.link_owner
+      # Creates Account and Owner arrays (calls Account and Owner self.all methods)
+      self.all
+      Bank::Owner.all
+      accounts_with_owners = []
+      account_owners_csv = CSV.read("support/account_owners.csv")
+      # for each row in the
+      account_owners_csv.each do |row|
+        account = self.find(row[0])
+        owner_account = Bank::Owner.find(row[1].to_i)
+        account.owner = owner_account
+        accounts_with_owners.push(account)
+      end
+      return accounts_with_owners
+    end
+
+    # ----------------------------------------- #
+              # Work below is extra #
+
+    # Displays (with formatting) the account details for all the accounts in accounts.csv
+    # (Made this on accident and didn't want to let it go to waste!)
+    def self.all_print_nice
+      if @accounts == nil
+        puts "There are no accounts."
+      else
+        @accounts.each do |account|
+          puts account
+          account.current_balance
+        end
+      end
+      return
+    end
+
+    # Finds an account by id passed in as a parameter and displays the account information nicely formatted
+    # (Made this on accident and didn't want to let it go to waste!)
+    def self.find_and_display(id_search)
+      found = @accounts.find do |account|
+        account.id == id_search.to_s
+      end
+      return found.current_balance
+    end
+          # end of extra work #
+# ----------------------------------------#
+
+    # Method for withdrawing from account
+    def withdraw(orig_amount_to_withdraw)
+      amount_to_withdraw = orig_amount_to_withdraw + @fee
+      # check if the requested withdrawal is more than the minimum balance
+      if (@balance - amount_to_withdraw) < @min_balance
+        # If we are dealing with a MoneyMarket account, we do special withdrawal behavior
+        # Keeping this here instead of in Money Market prevented us from having to rewrite an entire withdraw function for Money Market (we can just super the Account method)
+        if @acct_type == "MoneyMarket"
+          # Checks that the withdrawal won't put you below 0 or if you are no longer allowed to withdraw
+          if (@balance - amount_to_withdraw - @money_market_fee) < 0 || @allowed_to_withdraw == false
+            puts "Sorry you cannot make that withdrawal without depositing more money."
+          else
+            # If the withdrawal goes below min balance a fee is imposed and no further withdrawals are permitted
+            puts "As this transaction puts your balance below $#{@min_balance.round(2)}, a fee of $#{@money_market_fee.round(2)} has been imposed."
+            @balance -= (amount_to_withdraw + @money_market_fee)
+            puts "Your balance is $#{@balance.round(2)}"
```

Comment:
It seems like you use `round(2)` a lot so it might make sense to make a method that does this conversion for you rather than calling this method the same way repeatedly